src/pymor/algorithms/tr.py

```diff
@@ -0,0 +1,306 @@
+# This file is part of the pyMOR project (https://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (https://opensource.org/licenses/BSD-2-Clause)
+
+import warnings
+from copy import deepcopy
+
+import numpy as np
+
+from pymor.algorithms.bfgs import bfgs
+from pymor.core.base import BasicObject
+from pymor.core.defaults import defaults
+from pymor.core.exceptions import TRError
+from pymor.core.logger import getLogger
+from pymor.parameters.base import Mu
+
+
+class TRSurrogate(BasicObject):
+    """Surrogate for the :func:`trust_region.
+
+    Not intended to be used directly.
+    """
+
+    def __init__(self, reductor, initial_guess):
+        self.__auto_init(locals())
+
+        # generate a first rom if none was given
+        self.fom = reductor.fom
+        self.rom = None
+        if reductor._last_rom is None:
+            initial_basis = reductor.fom.solve(initial_guess)
+            reductor.extend_basis(initial_basis)
+            reductor.reduce()
+        self.rom = reductor._last_rom
```

Comment:
```suggestion
        self.rom = reductor.reduce()
```
and put the first enrichment in the TR algorithm for clarity?

Reply:
I wasn't quite sure about this. My thought was that adding the initial "extension" of the surrogate RB into the TR would add (unnecessary) clutter: after all, if the end user just wants to understand the TR algorithm they would not need to know that the surrogate is always enriched for the initial guess upon initialization without a reduced basis.

Reply:
Would it be a better idea to use the `extend` method here? Then the initial extension fits to the other extensions and the user only needs to adapt the method.

Reply:
I have now reduced the code in the `TRSurrogate` to

```python
# generate a first rom if none was given
self.fom = reductor.fom
self.extend(initial_guess)
self.accept()
```

this still keeps the initial extension outside of the TR but has to take the detour through `accept()` because of the TR logic linked to extensions.

Please let me know if this wasn't what you had intended.